File: catdog/CatDog.cpp
#include "CatDog.hpp"
#include <iostream>

CatDog::CatDog(const char* name) {
    // Why Animal without explicit initialization?
    this->name = name;
    std::cout << "CatDog()" << std::endl;
}
CatDog::~CatDog() {
    std::cout << "~CatDog()" << std::endl;
}

void CatDog::eat() {
    std::cout << this->getName() << " eats meat and fish..." << std::endl;
}

const char* CatDog::getName() const {
    return this->name;
}

File: java/catdog/CatDog.java
public class CatDog implements Cat, Dog {
    private String name;

    public CatDog() {
        this.name = "CatDog";
    }

    public void eat() {
        System.out.println(this.name + " eats fish and meat");
    }

    public void bark() {
        System.out.println(this.name + " say woof-woof...");
    }

    public void meow() {
        System.out.println(this.name + " say meow...");
    }
}